### PR TITLE
Feature/ros communication

### PR DESCRIPTION
**Pull request by TimBuckers**
All documentation on this PR and IPD in general is here: https://github.com/project-march/tutorials/pull/51/files. 

In the above documentation, 'the readme' is this readme. The readme link is correct after this PR is merged.

### CODE REVIEW

File: src/main.cpp [hunk 1]
```diff
@@ -78,15 +154,44 @@ void loop(){
...
+  {
     driver.go();
   }
 
+  if (received_gait_instruction_response)
+  {
+    // This means gait instruction handled
+    triggerPress = "EXIT_GAIT";
+    received_gait_instruction_response = false;
+  }
```
Comment (Ishadijcks) [inline src/main.cpp:166]: Why do you exit the gait once you receive a response?

File: src/StateMachine.cpp [hunk 2]
```diff
@@ -1,11 +1,29 @@
...
+    stateToGaitMapping[State::HomeSitActivated] = "home_sit";
+    stateToGaitMapping[State::SingleStepNormalActivated] = "gait_single_step";
+
+}
+
+State StateMachine::getCurrentState(){
+    return this->currentState;    
+}
+
+const char* StateMachine::getGaitName(State state){
```
Comment (Ishadijcks) [inline src/StateMachine.cpp]: Why not a string?

File: include/WirelessConnection.h [hunk 3]
```diff
@@ -0,0 +1,71 @@
+// Copyright 2018 Project March.
+
+#ifndef WII_NUNCHUK_UNO_ROS_WIFI_H
```
Comment (Ishadijcks) [inline include/WirelessConnection.h]: Nunchuck?

**Review by Ishadijcks — CHANGES_REQUESTED** [review]
Some small changes needed

**Comment by TimBuckers** [comment]
I think there is a bug (couldn't reproduce) walk start directly over again after closing step

File: src/main.cpp (hunk 1 above)
Comment (TimBuckers) [reply src/main.cpp:166]: To continue to the next screen on the ipd, this allows the pilot to go to the next gait already

**Review by Ishadijcks — APPROVED** [review]

**Pull request merged**